### PR TITLE
CalH5 Memo

### LINKED ISSUE #1424: CalH5 memo
We need to write up a proper description of CalH5, similar to the UVH5 memo.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Add a memo describing the CalH5 file format, along the lines of the UVH5 memo.

Also added several missing optional header items to the UVH5 memo and did some other doc updates.

## Motivation and Context

closes #1424

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Documentation change checklist:
- [x] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

**Comment by bhazelton** [comment]
I haven't yet added the pdfs to this PR since I figured there'd be updates needed, but I'm attaching them here to help with review.
[calh5.pdf](https://github.com/user-attachments/files/16381853/calh5.pdf)
[uvh5_memo.pdf](https://github.com/user-attachments/files/16381854/uvh5_memo.pdf)

**Review by kartographer — COMMENTED** [review]
I think this is looking good @bhazelton! One comment of moderate significance that I think should be fast to address, otherwise I think this is looking good to go.

File: docs/references/calh5.tex [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+any compression applied to a dataset, there is little that the user has to explicitly
+do when reading data. For users interested in creating new files, the use of
+compression is not strictly required by the CalH5 format, again because the
+HDF5 file is self-documenting in this regard, but compression is quite common.
+
+In the discussion below, we discuss required and optional datasets in the
+various groups. We note in parenthesis the corresponding attribute of a UVCal
+object. Note that in nearly all cases, the names are coincident, to make things
+as transparent as possible to the user.
+
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply docs/references/calh5.tex:97]: I added a paragraph in the introduction about this, let me know if it's sufficiently detailed.

File: docs/references/calh5.tex [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+encoded in the \textbf{wide\_band} parameter. All delay style calibration solutions
+(encoded in the \textbf{cal\_type} parameter) are wide band (since the calibration
+is calculated as a single time delay per antenna across a wide band) but gain style
+calibration solutions can also be wide band in some cases. For wide band calibration
+solutions, the frequency axis is replaced with a spectral window axis. The name of the
+primary calibration data array also depend on the \textbf{cal\_type}, it is complex-valued
+and called the \textbf{gain\_array} for gain style calibrations and is real valued and
+called the \textbf{delay\_array} for delay style calibrations. Another important difference
+is whether the calibration solutions are calculated per time integration or over a range
+of times. If it's calculated per integration, the solution times are stored in the
```
Comment (bhazelton) [inline docs/references/calh5.tex:60]: it's -> the solutions are

File: docs/references/calh5.tex [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+\textbf{time\_array} parameter, otherwise they are stored in the \textbf{time\_range}
+parameter, which has a second axis of length 2 to record the beginning and ending
+time for each time range. Finally, if there are some metadata parameters that are
```
Comment (bhazelton) [inline docs/references/calh5.tex:63]: Finally, if there are -> Finally, there are

**Review by bhazelton — COMMENTED** [review]
found a couple typos

**Review by kartographer — APPROVED** [review]
Looking good to me, thanks @bhazelton!

**Comment by bhazelton** [comment]
ok, I'll add the pdfs and then get you to re-review @kartographer

**Pull request merged**